cmd/operator/deploy/operator/10-collector.yaml

```diff
@@ -0,0 +1,154 @@
+        - --export.user-agent=prometheus/2.35.0-gmp.1 (mode:kubectl)
+        # The environment variable EXTRA_ARGS will be populated by the operator.
+        # DO NOT specify it here.
+        env:
+        - name: GOGC
+          value: "25"
+        ports:
+        - name: prom-metrics
+          containerPort: 19090
+        livenessProbe:
+          httpGet:
+            path: /-/healthy
+            port: 19090
+            scheme: HTTP
+        readinessProbe:
+          httpGet:
+            path: /-/ready
+            port: 19090
+            scheme: HTTP
+        resources:
+          limits:
+            cpu: 1000m
+            memory: 3G
+          requests:
+            cpu: 100m
+            memory: 200M
+        volumeMounts:
+        - name: storage
+          mountPath: /prometheus/data
+        - name: config-out
+          mountPath: /prometheus/config_out
+          readOnly: true
+        - name: rules-secret
+          mountPath: /etc/secrets
+          readOnly: true
+        securityContext:
+          allowPrivilegeEscalation: false
+          privileged: false
+          capabilities:
+            drop:
+            - all
+          runAsUser: 1000
+          runAsGroup: 1000
+          seccompProfile:
+            type: RuntimeDefault
+      - name: config-reloader
+        image: gke.gcr.io/prometheus-engine/config-reloader:v0.4.0-gke.0
+        args:
+        - --config-file=/prometheus/config/config.yaml
+        - --config-file-output=/prometheus/config_out/config.yaml
+        - --reload-url=http://localhost:19090/-/reload
+        - --listen-address=:19091
+        env:
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+              fieldPath: spec.nodeName
+        ports:
+        - name: cfg-rel-metrics
+          containerPort: 19091
+        resources:
+          limits:
+            cpu: 10m
+            memory: 32M
+          requests:
+            cpu: 5m
+            memory: 16M
+        volumeMounts:
+        - name: config
+          mountPath: /prometheus/config
+          readOnly: true
+        - name: config-out
+          mountPath: /prometheus/config_out
+        securityContext:
+          allowPrivilegeEscalation: false
+          privileged: false
+          capabilities:
+            drop:
+            - all
+          runAsUser: 1000
+          runAsGroup: 1000
+          seccompProfile:
+            type: RuntimeDefault
+      serviceAccountName: collector
+      automountServiceAccountToken: true
+      priorityClassName: gmp-critical
+      tolerations:
+      - effect: NoExecute
+        operator: Exists
+      - effect: NoSchedule
+        operator: Exists
+      volumes:
+      - name: storage
+        emptyDir: {}
+      - name: config
+        configMap:
+          name: collector
+      - name: config-out
+        emptyDir: {}
+      - name: rules-secret
+        secret:
+          secretName: collection
```

Comment:
That sounds sensible? I think this can be done in a follow-up without adverse effects as this change is already fairly big. (Probably also should come with further pruning RBACs on those secerts etc.)